libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -684,6 +684,57 @@ namespace webassembly {
          */
          void set_action_return_value(span<const char> packed_blob);
 
+         /**
+          * Makes a sync call in the context of the parent action or parent sync call
+          * of this operation.
+          *
+          * @ingroup sync call
+          * @param receiver - the name of the account that the sync call is made to.
+          * @param flags - flags (bits) representing blockchain level requirements about
+          *                the sync call. LSB bit indicates read-only. All other bits
+          *                are reserved to be 0.
+          * @param data - the data of the sync call, in the format of
+          *
+          *               struct data_header {
+          *                  uint32_t version        = 0;
+          *                  uint64_t function_name;
+          *
+          *                  // Flags representing function level requirements
+          *                  // about the sync call. LSB of flags indicates no_op_if_not_defined.
+          *                  // All other bits are reserved to be 0.
+          *                  // If no_op_if_not_defined is set, the sync call is no-op
+          *                  // if the receiver contract does not have sync_call entry
+          *                  // point or the signature does not match the format,
+          *                  // or the called function_name is not recognized by the receiver.
+          *                  // Otherwise, the call aborts.
+          *                  uint64_t flags          = 0;
+          *               };
+          *
+          *               struct data_t {
+          *                  data_header      header;
+          *                  span<const char> arguments;
+          *               };
```

Comment:
Is this the appropriate location for the `data` documentation? The protocol is agnostic to what `data` is, so nodeos doesn't care... until output is shown in any API calls I guess. So I'm not sure..

Reply:
I was thinking about adding something like this is a recommended data format or updated CDT expects to use this. You are correct, this is not the right place to specify it. Spring does not care `data`. Will remove it.